### DISCUSSION TITLE
Enforce semantic PR title

### DISCUSSION THREAD

**Discussion by ad1992** [excalidraw/excalidraw#2531] [General]
As now Excalidraw is a monorepo so it would be better to have semantic PR titles that can be used in the changelog and later can also help in automating the entire release process.

Since we follow `squash and merge` so the pr title ultimately gets used as the commit message so it would also help in deciding whether the next version should be patch/major/breaking version when looking at the changelog. (This can be later automated as well)

We can  start using [https://github.com/zeke/semantic-pull-requests](https://github.com/zeke/semantic-pull-requests) which also has a GitHub action [https://github.com/amannn/action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request)

**Comment by lipis** [comment]
[Suggested types](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json):

```
  - feat
  - fix
  - docs
  - style
  - refactor
  - perf
  - test
  - build
  - ci
  - chore
  - revert
```

**Comment by OrenMe** [comment]
We use this in all of our repos and it makes the release verse UI int and release notes real easy and nice.
Totally in favor.

**Comment by lipis** [comment]
We added GitHub actions to enforce them.
